src/derivations/Segmentation.js

```diff
@@ -405,7 +405,9 @@ export default class Segmentation extends DerivedPixels {
         const reNumberedSegmentCopy = {
             SegmentedPropertyCategoryCodeSequence:
                 Segment.SegmentedPropertyCategoryCodeSequence,
-            SegmentNumber: (SegmentSequence.length + 1).toString(),
+            SegmentNumber:
+                Segment.SegmentNumber ||
+                (SegmentSequence.length + 1).toString(),
```

Comment:
hmmmmm, I encountered this when I was writing the segmentation export and my code was generating a fake segmentation with two segments of 2 and 5 and I saw they get mapped into different color when I re import them (since they were getting mapped to 1 and 2) but now that you sent that link maybe you are right. So does this mean there is no way to specifically export a segmentation object that specifically number the segments arbitrary? interesting I guess

Reply:
> there is no way to specifically export a segmentation object that specifically number the segments arbitrary?

Correct. But why this would be an issue? Segment number is just a pointer within the segmentation object to connect semantics of the segment and the color of the segment with the corresponding pixel data.

Reply:
The issue I encountered initially was the color lut for segments (colors gets changed before export and after import), I think we have a bug in OHIF too ...

Something strange I saw in SEG export was this, not sure if there is an explanation for it 

My metadata for dcmjs to export SEG using this rgb2DICOMLAB 

```js
const RecommendedDisplayCIELabValue = dcmjs.data.Colors.rgb2DICOMLAB(
        color.slice(0, 3)
    );
```

![image](https://github.com/dcmjs-org/dcmjs/assets/7490180/f881d2ad-db82-4eb8-9394-47dfc22c7cdd)


But later looking at the DICOM again uing the DICOM dump [here](https://rawgit.com/cornerstonejs/dicomParser/master/examples/dumpWithDataDictionary/index.html)

![image](https://github.com/dcmjs-org/dcmjs/assets/7490180/424a7573-bcd7-4f28-9824-35f9865f5ede)

As you see the encoded results are different, so my parser with `dicomlab2RGB` returns a different color. 

Am I missing an obvious thing here?

Reply:
I don't think I was involved in writing that conversion function.

But in dcmqi we did spend a bit of time debugging it, and in the end the conversion was implemented in DCMTK. Maybe this can help debug OHIF conversion routines: https://github.com/DCMTK/dcmtk/blob/master/dcmiod/libsrc/cielabutil.cc.

Reply:
The display values are clearly out of range - from the spec:

Attributes such as Graphic Layer Recommended Display CIELab Value (0070,0401) consist of three unsigned short values:

An L value linearly scaled to 16 bits, such that 0x0000 corresponds to an L of 0.0, and 0xFFFF corresponds to an L of 100.0.

An a* then a b* value, each linearly scaled to 16 bits and offset to an unsigned range, such that 0x0000 corresponds to an a* or b* of -128.0, 0x8080 corresponds to an a* or b* of 0.0 and 0xFFFF corresponds to an a* or b* of 127.0

So, the actual values of L* a* and b* for unencoded are:
0...100, -128...127, -128...127
or the encoded values are:
0...65535
for PCS encoding.  The provided values are completely outside the range, so something is converting things incorrectly before being passed into dcmjs

Reply:
Ok silly mistake, apparently the RGB that rgb2DICOMLAB accepts is scaled to 0-1 RGB, my bad